### PR TITLE
[DROOLS-6316] Refactor DefineFunctionUtils/ExpressionFunctionUtils

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @kostola  @jiripetrlik

See https://issues.redhat.com/browse/DROOLS-6316

This refactoring aims at
1) put/move most "Expression" related codegen inside ExpressionFunctionUtils
2) use the "template" approach as much as possible
3) better define the methods inside ExpressionFunctionUtils

### CODE REVIEW

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        switch (this) {
+            case PLUS:
+                return plus(inputData);
+            default:
+                throw new KiePMMLException("Unmanaged BUILTIN_FUNCTIONS " + this);
+        }
+    }
+
+    private double plus(final Object[] inputData) {
+        if (inputData.length < 2) {
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java]: Shouldn't we rather check that the length is equal to 2? Is there any reason for more operands? If yes, we are still doing the sum of only two operands.

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/resources/ExpressionFunctionUtilsTemplate.tmpl [hunk 2]
```diff
@@ -0,0 +1,35 @@
+public class ExpressionFunctionUtilsTemplate {
+
```
Comment (jiripetrlik) [inline kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/resources/ExpressionFunctionUtilsTemplate.tmpl:2]: Unexpected new lines.

**Review by jiripetrlik — COMMENTED** [review]
Thank you @gitgabrio for PR. I have two comments.

File: kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java (hunk 1 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-api/src/main/java/org/kie/pmml/api/enums/BUILTIN_FUNCTIONS.java]: @jiripetrlik done, thanks for spotting that! I took the chance to add a testing class

File: kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/resources/ExpressionFunctionUtilsTemplate.tmpl (hunk 2 above)
Comment (gitgabrio) [reply kie-pmml-trusty/kie-pmml-compiler/kie-pmml-compiler-commons/src/main/resources/ExpressionFunctionUtilsTemplate.tmpl:2]: @jiripetrlik done

**Comment by gitgabrio** [comment]
Jenkins retest this

**Review by jiripetrlik — APPROVED** [review]
@gitgabrio
Thank you for the changes.

**Comment by gitgabrio** [comment]
Jenkins run cdb

**Comment by gitgabrio** [comment]
Jenkins retest this

**Review by danielezonca — APPROVED** [review]

**Review by kostola — APPROVED** [review]

**Comment by gitgabrio** [comment]
Jenkins retest this

**Pull request merged**